core/Pos/Util/Util.hs

```diff
@@ -260,6 +261,10 @@ type instance ChannelT (Ether.TaggedTrans tag t m) = ChannelT m
 maybeThrow :: (MonadThrow m, Exception e) => e -> Maybe a -> m a
 maybeThrow e = maybe (throwM e) pure
 
+-- | Fail or return result depending on what is stored in 'Either'.
+eitherToFail :: (MonadFail m, ToString s) => Either s a -> m a
```

Comment:
o, fuck.
it's the wrong place to write a comment.

I noticed that you used `MonadError` somewhere, and don't get why did you use it